JetBrains: Make it compatible with 2022.2

vdavid opened the pull request:
Fixes https://github.com/sourcegraph/sourcegraph/issues/39571

- Updated platform+Java+IntelliJ versions
- Fixed the theme bug: apparently, JetBrains started using a new color class
- Bumped plugin version to 2.0.0, to go ahead with the release

## Test plan

Tested it, it looks nice again. When changing the theme, the “panda bug” reappears, but it's not directly related to this bug. I've created a separate issue for this, with priority:high, but in the Icebox: https://github.com/sourcegraph/sourcegraph/issues/39574

## App preview:

- [Web](https://sg-web-dv-jetbrains-2022-2-compatibility.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-jwfngpyduk.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

philipp-spiess commented on client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java:
```diff
@@ -26,7 +27,7 @@ public static JsonObject getCurrentThemeAsJson() {
             try {
                 Object value = UIManager.get(key);
-                if (value instanceof ColorUIResource) {
+                if (value instanceof ColorUIResource || value instanceof LafManagerImpl.IJColorUIResource) {
```
🤦

philipp-spiess commented on client/jetbrains/gradle.properties:
```diff
@@ -3,7 +3,7 @@
 pluginName=Sourcegraph
 # SemVer format -> https://semver.org
-pluginVersion=2.0.0-alpha.5
+pluginVersion=2.0.0
```
🚢

philipp-spiess reviewed (approved):
We also need to update the CHANGELOG I believe 🙂

philipp-spiess commented on client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java:
```diff
@@ -26,7 +27,7 @@ public static JsonObject getCurrentThemeAsJson() {
             try {
                 Object value = UIManager.get(key);
-                if (value instanceof ColorUIResource) {
+                if (value instanceof ColorUIResource || value instanceof LafManagerImpl.IJColorUIResource) {
```
Does `LafManagerImpl.IJColorUIResource` exist on lower IntelliJ versions? I.e have you tried running the extension on 2021.1?

philipp-spiess replied:
![image](https://user-images.githubusercontent.com/458591/181510784-7c417a44-4048-4a61-9544-f008b3b25d39.png)

Oh no

vdavid commented:
> We also need to update the CHANGELOG I believe 🙂

CHANGELOG is always a pain in the ass for me because I never know the format. I've changed `[Unreleased]` to `[2.0.0]` and added some details. Do you know if this is the right format @philipp-spiess?

philipp-spiess commented on client/jetbrains/src/main/java/com/sourcegraph/config/ThemeUtil.java:
```diff
@@ -26,7 +25,7 @@ public static JsonObject getCurrentThemeAsJson() {
             try {
                 Object value = UIManager.get(key);
-                if (value instanceof ColorUIResource) {
+                if (value instanceof Color) {
```
Great find!

philipp-spiess reviewed (approved):
Ship it!

Verified that it works on 2021.1 on Windows:

![Screenshot 2022-07-28 at 15 41 12](https://user-images.githubusercontent.com/458591/181520016-8244143f-b816-4c8d-b484-18b27a89d70a.png)
